In [2]:
!nmap scanme.org

/bin/bash: line 1: nmap: command not found


In [3]:
!apt-get install nmap -y

Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following additional packages will be installed:
  liblinear4 liblua5.3-0 libpcap0.8 lua-lpeg nmap-common
Suggested packages:
  liblinear-tools liblinear-dev ncat ndiff zenmap
The following NEW packages will be installed:
  liblinear4 liblua5.3-0 libpcap0.8 lua-lpeg nmap nmap-common
0 upgraded, 6 newly installed, 0 to remove and 2 not upgraded.
Need to get 6,029 kB of archives.
After this operation, 26.5 MB of additional disk space will be used.
Get:1 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 libpcap0.8 amd64 1.10.1-4ubuntu1.22.04.1 [145 kB]
Get:2 http://archive.ubuntu.com/ubuntu jammy/universe amd64 liblinear4 amd64 2.3.0+dfsg-5 [41.4 kB]
Get:3 http://archive.ubuntu.com/ubuntu jammy/main amd64 liblua5.3-0 amd64 5.3.6-1build1 [140 kB]
Get:4 http://archive.ubuntu.com/ubuntu jammy/universe amd64 lua-lpeg amd64 1.0.2-1 [31.4 kB]
Get:5 http://archive.ubuntu.com/ubuntu jammy-

In [4]:
!nmap scanme.nmap.org
#This command give the results of open and closed parts.

Starting Nmap 7.80 ( https://nmap.org ) at 2026-03-02 06:03 UTC
Nmap scan report for scanme.nmap.org (45.33.32.156)
Host is up (0.14s latency).
Other addresses for scanme.nmap.org (not scanned): 2600:3c01::f03c:91ff:fe18:bb2f
Not shown: 996 closed ports
PORT      STATE SERVICE
22/tcp    open  ssh
80/tcp    open  http
9929/tcp  open  nping-echo
31337/tcp open  Elite

Nmap done: 1 IP address (1 host up) scanned in 2.71 seconds


In [5]:
!nmap -F scanme.nmap.org
#Fast Scan, scan fewer ports than default.

Starting Nmap 7.80 ( https://nmap.org ) at 2026-03-02 06:03 UTC
Nmap scan report for scanme.nmap.org (45.33.32.156)
Host is up (0.14s latency).
Other addresses for scanme.nmap.org (not scanned): 2600:3c01::f03c:91ff:fe18:bb2f
Not shown: 98 closed ports
PORT   STATE SERVICE
22/tcp open  ssh
80/tcp open  http

Nmap done: 1 IP address (1 host up) scanned in 0.79 seconds


In [7]:
!nmap -p 22,80 scanme.org
#This command scans the target for the particular ports that is for ssh and  http.

Starting Nmap 7.80 ( https://nmap.org ) at 2026-03-02 06:04 UTC
Nmap scan report for scanme.org (45.33.32.156)
Host is up (0.14s latency).
Other addresses for scanme.org (not scanned): 2600:3c01::f03c:91ff:fe18:bb2f
rDNS record for 45.33.32.156: scanme.nmap.org

PORT   STATE SERVICE
22/tcp open  ssh
80/tcp open  http

Nmap done: 1 IP address (1 host up) scanned in 0.61 seconds


In [8]:
!nmap -p 1-100 scanme.org:
#This command scans the target against the range of ports that is 1-100

Starting Nmap 7.80 ( https://nmap.org ) at 2026-03-02 06:05 UTC
Failed to resolve "scanme.org:".
Nmap done: 0 IP addresses (0 hosts up) scanned in 0.04 seconds


In [9]:
!nmap -T3 scanme.org
# -T3  :Controls scan speed & stealth
   #    Time values can be vary from 0-5

Starting Nmap 7.80 ( https://nmap.org ) at 2026-03-02 06:05 UTC
Nmap scan report for scanme.org (45.33.32.156)
Host is up (0.14s latency).
Other addresses for scanme.org (not scanned): 2600:3c01::f03c:91ff:fe18:bb2f
rDNS record for 45.33.32.156: scanme.nmap.org
Not shown: 996 closed ports
PORT      STATE SERVICE
22/tcp    open  ssh
80/tcp    open  http
9929/tcp  open  nping-echo
31337/tcp open  Elite

Nmap done: 1 IP address (1 host up) scanned in 2.64 seconds


In [11]:
!nmap -F scanme.org
#Fast Scan, scan fewer ports than default.

Starting Nmap 7.80 ( https://nmap.org ) at 2026-03-02 06:06 UTC
Nmap scan report for scanme.org (45.33.32.156)
Host is up (0.14s latency).
Other addresses for scanme.org (not scanned): 2600:3c01::f03c:91ff:fe18:bb2f
rDNS record for 45.33.32.156: scanme.nmap.org
Not shown: 98 closed ports
PORT   STATE SERVICE
22/tcp open  ssh
80/tcp open  http

Nmap done: 1 IP address (1 host up) scanned in 0.79 seconds


In [12]:
!nmap -sV scanme.org
# These gives version of the applications that is running in the target.

Starting Nmap 7.80 ( https://nmap.org ) at 2026-03-02 06:06 UTC
Nmap scan report for scanme.org (45.33.32.156)
Host is up (0.14s latency).
Other addresses for scanme.org (not scanned): 2600:3c01::f03c:91ff:fe18:bb2f
rDNS record for 45.33.32.156: scanme.nmap.org
Not shown: 996 closed ports
PORT      STATE SERVICE    VERSION
22/tcp    open  tcpwrapped
80/tcp    open  http       Apache httpd 2.4.7 ((Ubuntu))
9929/tcp  open  nping-echo Nping echo
31337/tcp open  tcpwrapped

Service detection performed. Please report any incorrect results at https://nmap.org/submit/ .
Nmap done: 1 IP address (1 host up) scanned in 10.54 seconds


In [13]:
!nmap -O scanme.nmap.org
#Detects Operating system of the Target machine

Starting Nmap 7.80 ( https://nmap.org ) at 2026-03-02 06:06 UTC
Nmap scan report for scanme.nmap.org (45.33.32.156)
Host is up (0.14s latency).
Other addresses for scanme.nmap.org (not scanned): 2600:3c01::f03c:91ff:fe18:bb2f
Not shown: 996 closed ports
PORT      STATE SERVICE
22/tcp    open  ssh
80/tcp    open  http
9929/tcp  open  nping-echo
31337/tcp open  Elite
Aggressive OS guesses: Linux 2.6.32 - 3.13 (96%), Linux 2.6.22 - 2.6.36 (95%), Linux 3.10 - 4.11 (94%), Linux 3.2 - 4.9 (94%), Linux 3.10 (94%), Linux 2.6.32 - 3.10 (93%), HP P2000 G3 NAS device (93%), Linux 2.6.18 (93%), Linux 3.16 - 4.6 (93%), Linux 2.6.32 (93%)
No exact OS matches for host (test conditions non-ideal).
Network Distance: 11 hops

OS detection performed. Please report any incorrect results at https://nmap.org/submit/ .
Nmap done: 1 IP address (1 host up) scanned in 7.02 seconds


VIRUS TOTAL FINDINGS

In [17]:

import requests
from google.colab import userdata

API_KEY = userdata.get("API_KEY")

url = "https://www.virustotal.com/api/v3/domains/scanme.nmap.org"

headers = {
    "x-apikey": API_KEY
}

response = requests.get(url, headers=headers)

print(response.status_code)
print(response.json())

200
{'data': {'id': 'scanme.nmap.org', 'type': 'domain', 'links': {'self': 'https://www.virustotal.com/api/v3/domains/scanme.nmap.org'}, 'attributes': {'total_votes': {'harmless': 1, 'malicious': 0}, 'whois': 'Creation Date: 1999-01-18T05:00:00Z\nDNSSEC: unsigned\nDomain Name: nmap.org\nDomain Status: clientTransferProhibited https://icann.org/epp#clientTransferProhibited\nName Server: ns1.linode.com\nName Server: ns2.linode.com\nName Server: ns3.linode.com\nName Server: ns4.linode.com\nName Server: ns5.linode.com\nRegistrar Abuse Contact Email: abuse@dynadot.com\nRegistrar Abuse Contact Phone: +1.6502620100\nRegistrar IANA ID: 472\nRegistrar URL: http://www.dynadot.com\nRegistrar WHOIS Server: whois.dynadot.com\nRegistrar: Dynadot Inc\nRegistry Domain ID: REDACTED\nRegistry Expiry Date: 2029-01-18T05:00:00Z\nUpdated Date: 2023-08-31T05:05:15Z', 'tld': 'org', 'last_analysis_stats': {'malicious': 0, 'suspicious': 0, 'undetected': 33, 'harmless': 60, 'timeout': 0}, 'popularity_ranks': {'

In [18]:
data = response.json()
attributes = data["data"]["attributes"]

In [19]:
print("Reputation Score:", attributes["reputation"])

Reputation Score: 1


In [20]:
print("Last Analysis Stats:")
print(attributes["last_analysis_stats"])

Last Analysis Stats:
{'malicious': 0, 'suspicious': 0, 'undetected': 33, 'harmless': 60, 'timeout': 0}


In [21]:
print("Categories:")
print(attributes["categories"])

Categories:
{'BitDefender': 'advice', 'Forcepoint ThreatSeeker': 'information technology', 'Sophos': 'information technology', 'Xcitium Verdict Cloud': 'downloads', 'alphaMountain.ai': 'Business/Economy, Information Technology, Information/Computer Security (alphaMountain.ai)'}


In [22]:
print("Last DNS Records:")
print(attributes["last_dns_records"])

Last DNS Records:
[{'type': 'A', 'ttl': 3600, 'value': '45.33.32.156'}, {'type': 'AAAA', 'ttl': 400, 'value': '2600:3c01::f03c:91ff:fe18:bb2f'}]


In [23]:
def get_data(endpoint):
    url = f"{BASE_URL}/{endpoint}"
    response = requests.get(url, headers=headers)

    if response.status_code != 200:
        print("Error:", response.text)
        return None

    return response.json()

Port & Service Analysis

In [27]:
import xml.etree.ElementTree as ET
tree = ET.parse("investigation.xml")
root = tree.getroot()

In [28]:
for host in root.findall("host"):
    for port in host.findall(".//port"):
        port_id = port.get("portid")
        service = port.find("service")

        if service is not None:
            service_name = service.get("name")
            print(port_id, ":", service_name)

22 : ssh
53 : domain
80 : http
9929 : nping-echo
31337 : tcpwrapped


In [29]:
for host in root.findall("host"):
    for port in host.findall(".//port"):
        service = port.find("service")

        if service is not None:
            name = service.get("name")
            version = service.get("version")

            print("Service:", name, "| Version:", version)

Service: ssh | Version: 6.6.1p1 Ubuntu 2ubuntu2.13
Service: domain | Version: None
Service: http | Version: 2.4.7
Service: nping-echo | Version: None
Service: tcpwrapped | Version: None


In [30]:
risky_services = ["ssh", "ftp", "mysql", "smb"]

for host in root.findall("host"):
    for port in host.findall(".//port"):
        service = port.find("service")

        if service is not None:
            name = service.get("name")

            if name in risky_services:
                print("Sensitive Service Detected:", name)

Sensitive Service Detected: ssh


Vulnerability Reasoning

1. Port 22 – SSH:
This port is used for remote login. If it is open to the public, attackers may try to guess passwords and break into the system.

2. Port 80 – HTTP:
This port is used for websites. Since it is not encrypted, data sent through it can be seen by others. If the website software is old, it may have security weaknesses.

3. Port 53 – DNS:
This port helps convert domain names into IP addresses. If not properly protected, attackers may misuse it to gather information or overload the server.

4. Port 9929 – nping-echo:
This is usually a testing service. If left open, it may give information about the network that could help attackers understand the system better.

5. Port 31337 – tcpwrapped:
This means the port is protected by some security control, like a firewall. Even though it is protected, it still shows that something is running behind it.


CONCLUSION

In this exercise, I learned how to scan a system and understand which services are running on it. I understood that open ports show which services are available and that each open service can create a possible security risk if not properly protected.

By analyzing the ports and services, I learned how attackers may look at a system to find weak points. I also understood the importance of securing services like SSH and web servers to reduce security risks.

Overall, this assignment helped me understand how network scanning works and how to think about security risks without actually exploiting the system. It improved my understanding of basic cybersecurity concepts like ports, services, and vulnerability reasoning.

